src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -0,0 +1,71 @@
+import { useEffect, useState } from 'react';
+import { useDispatch } from 'react-redux';
+import { actions as todosActions } from '../../store/todos';
+
+export const TodoFilter: React.FC = () => {
+  const dispatch = useDispatch();
+  const [status, setStutus] = useState('');
+  const [query, setQuery] = useState('');
+  const reset = () => {
+    setQuery('');
+  };
+
+  useEffect(() => {
+    switch (status) {
+      case 'completed':
+        dispatch(todosActions.filterByCompletedTodos(query));
+        break;
+
+      case 'active':
+        dispatch(todosActions.filterByActiveTodos(query));
+        break;
+
+      default:
+        dispatch(todosActions.filterAllTodos(query));
```

Comment:
If you have a specific result for the specific case it's better to clarify this case, even if it's the same as default.
Also you can use enum for cases and options in select